modules/navier_stokes/src/physics/WCNSFVFlowPhysics.C

```diff
@@ -1349,6 +1384,51 @@ WCNSFVFlowPhysics::getPorosityFunctorName(bool smoothed) const
     return _porosity_name;
 }
 
+MooseFunctorName
+WCNSFVFlowPhysics::getLinearFrictionCoefName() const
+{
+  // Check all blocks. If more than one block, they would need to be consolidated #include in
+  // a single functor material. We won't implement this for now
+  if (_friction_types.empty())
+    return "0";
+  else if (_friction_types.size() == 1)
+  {
+    for (const auto & type_i : index_range(_friction_types[0]))
+    {
+      const auto upper_name = MooseUtils::toUpper(_friction_types[0][type_i]);
+      if (upper_name == "DARCY")
+        return _friction_coeffs[0][type_i];
+    }
+    // No linear type found
+    return "0";
+  }
+  else if (_friction_types.size() > 1)
+  {
+    bool linear_friction_factor_found = false;
+    MooseFunctorName linear_friction_factor;
+    for (const auto block_i : index_range(_friction_types))
+      for (const auto & type_i : index_range(_friction_types[0]))
```

Comment:
Is there a reason why you keep indexing at 0 instead of with `block_i`?
```suggestion
      for (const auto type_i : index_range(_friction_types[block_i]))
```

Reply:
no it s a mistake. [0] would be fine if all blocks had the same types active